src/internal/packager2/pull.go

```diff
@@ -120,9 +123,10 @@ func pullOCI(ctx context.Context, src, tarDir, shasum string, filter filters.Com
 	if shasum != "" {
 		src = fmt.Sprintf("%s@sha256:%s", src, shasum)
 	}
+	// TODO use layout.NewRemote and create layout.Pull
 	arch := config.GetArch()
 	platform := oci.PlatformForArch(arch)
```

Comment:
This `config.GetArch` call is only used for a platform for logging. Since both of the callers of `pullOCI` already call `config.GetArch` we can delete it and replace the `platformForArch` call with the passed in architecture.

Reply:
Updated